dev/pyproject.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import re
+from pathlib import Path
+
+import toml
+
+SEPARATOR = """
+# Package metadata: can't be updated manually, use dev/pyproject.py
+# -----------------------------------------------------------------
+# Dev tool settings: can be updated manually
+
+"""
+
+
+def read_requirements(path: Path) -> list[str]:
+    lines = (l.strip() for l in path.read_text().splitlines())
+    return [l for l in lines if l and not l.startswith("#")]
+
+
+def build(skinny: bool) -> None:
+    skinny_requirements = read_requirements(Path("requirements", "skinny-requirements.txt"))
+    core_requirements = read_requirements(Path("requirements", "core-requirements.txt"))
+    gateways_requirements = read_requirements(Path("requirements", "gateway-requirements.txt"))
+    version = re.search(
+        r'^VERSION = "([a-z0-9\.]+)"$', Path("mlflow", "version.py").read_text(), re.MULTILINE
+    ).group(1)
+    python_version = Path("requirements", "python-version.txt").read_text().strip()
+    data = {
+        "build-system": {
+            "requires": ["setuptools"],
+            "build-backend": "setuptools.build_meta",
+        },
+        "project": {
+            "name": "mlflow" if not skinny else "mlflow-skinny",
+            "version": version,
+            "authors": [{"name": "Databricks", "email": "databricks@databricks.com"}],
```

Comment:
Can we change this to the maintainers group email address?

Reply:
Sounds good. I'll use maintainers instead of authors.